distributed-job-manager/src/main/java/ru/fix/distributed/job/manager/Worker.java

```diff
@@ -318,11 +318,7 @@ private void updateWorkPoolForJob(DistributedJob job, Set<String> newWorkPool) {
                         boolean workerAliveFlagPathExists = null != transaction.checkPath(workerAliveFlagPath);
 
                         if (workPoolsPathExists && availableJobPathExists && workerAliveFlagPathExists) {
-                            boolean workPoolUpdated = workPoolSubTree.updateJob(transaction, jobId, newWorkPool);
-                            if (workPoolUpdated) {
-                                transaction.setData(paths.aliveWorker(workerId),
-                                        curatorFramework.getData().forPath(workerAliveFlagPath));
-                            }
```

Comment:
As we can see, "dynamic work-pool" (https://github.com/ru-fix/distributed-job-manager/issues/16) is already implemented by updating worker-alive node. Despite that, i think its better to make manager listen to work-pool tree. It will make DJM work more clear